### PR TITLE
Payments blocks: Add more robust set of keywords

### LINKED ISSUE #22898: Donations Block: Update the keywords
### Impacted plugin

Jetpack

### What

Add more keywords to the Donations block

### How

Currently, the Donations block only has one keyword: `Donations`.
This is not particularly useful, as the matcher would fail even with a simple `/donate` search.

Other related blocks have many more keywords. Let's take inspiration from them and add more keywords to the Donations block!

- Payments
  - sell
  - subscriptions
  - product
  - memberships
- Pay with PayPal
  - buy
  - commerce
  - products
  - purchase
  - sell
  - shop
  - simple
  - payments

### PR DESCRIPTION
**Pull request by dsas**
Fixes #22898

#### Changes proposed in this Pull Request:

Add a more extensive set of keywords to some of the payments blocks, to
make it easier for users to find what they are looking for.

#### Jetpack product discussion

See paYJgx-1Hh-p2#comment-2156

#### Does this pull request change what data or activity we track or use?

No

##### Setup
**jetpack**: Build 
**wpcom**: Patch sandbox as usual
**Atomic**: Use jetpack beta plugin to select this branch

##### Test 
 1. Open a new post
 2. Type the keywords
 3. See blocks appear

### CODE REVIEW

**Review by jeherve — COMMENTED** [review]
That seems like a lot. Is there a limit we have to keep in mind with keywords? I remember that at some point there was a limit of 3 or 4 keywords, but maybe it's been lifted now?

**Comment by dsas** [comment]
> That seems like a lot. Is there a limit we have to keep in mind with keywords?

@jeherve [the docs say unlimited](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-metadata/#keywords) Whether I search for keywords at the beginning, end of middle of the list they all work.

I can rowback on some of the more niche words if you think this is overzealous.

**Review by jeherve — COMMENTED** [review]
> I can rowback on some of the more niche words if you think this is overzealous.

If there are no limits, I suppose we can have more keywords. 

That said, it does seem like generic terms like `accept`, `give`, `giving`, `gratuity`, `money`, `onetime`, `one time`, `repeat`, `toast`, `tipping`, `support`, `yearly` seem too generic for the Donations block for example, imo. @mtias What's your take on this?

**Comment by Copons** [comment]
cc @sethmclellana8c

**Comment by dsas** [comment]
I think `money` is fair game, these blocks are all directly about taking money via a website. I've removed a few dozen other keywords though.

**Comment by mtias** [comment]
This looks a bit intense, I'd suggest starting smaller and going from there.

File: projects/plugins/jetpack/extensions/blocks/donations/index.js [hunk 1]
```diff
@@ -24,7 +24,38 @@ export const settings = {
 	description: __( 'Collect one-time, monthly, or annually recurring donations.', 'jetpack' ),
 	icon: DonationsIcon,
 	category: 'earn',
-	keywords: [ __( 'Donations', 'jetpack' ) ],
+	keywords: [
+		_x( 'charity', 'block search term', 'jetpack' ),
+		_x( 'contribution', 'block search term', 'jetpack' ),
+		_x( 'commerce', 'block search term', 'jetpack' ),
```
Comment (mtias) [inline projects/plugins/jetpack/extensions/blocks/donations/index.js:30]: ecommerce should already be covering this one

File: projects/plugins/jetpack/extensions/blocks/donations/index.js [hunk 2]
```diff
@@ -24,7 +24,38 @@ export const settings = {
...
+		_x( 'credit card', 'block search term', 'jetpack' ),
+		_x( 'debit card', 'block search term', 'jetpack' ),
+		_x( 'donate', 'block search term', 'jetpack' ),
+		_x( 'Donations', 'block search term', 'jetpack' ),
```
Comment (mtias) [inline projects/plugins/jetpack/extensions/blocks/donations/index.js:34]: The block title should already be covering this one

File: projects/plugins/jetpack/extensions/blocks/donations/index.js [hunk 3]
```diff
@@ -24,7 +24,38 @@ export const settings = {
...
+		_x( 'earn', 'block search term', 'jetpack' ),
+		_x( 'ecommerce', 'block search term', 'jetpack' ),
+		_x( 'fundraising', 'block search term', 'jetpack' ),
+		_x( 'fundraiser', 'block search term', 'jetpack' ),
+		'gofundme',
+		'go fund me',
```
Comment (mtias) [inline projects/plugins/jetpack/extensions/blocks/donations/index.js:40]: This one should be removed, it'd match individual generic words and doesn't make sense

File: projects/plugins/jetpack/extensions/blocks/donations/index.js [hunk 4]
```diff
@@ -24,7 +24,38 @@ export const settings = {
...
+		_x( 'memberships', 'block search term', 'jetpack' ),
```
Comment (mtias) [inline projects/plugins/jetpack/extensions/blocks/donations/index.js:41]: Not sure this one makes sense given the role of the block

File: projects/plugins/jetpack/extensions/blocks/donations/index.js [hunk 5]
```diff
@@ -24,7 +24,38 @@ export const settings = {
...
+		_x( 'monthly', 'block search term', 'jetpack' ),
+		_x( 'nonprofit', 'block search term', 'jetpack' ),
+		_x( 'non-profit', 'block search term', 'jetpack' ),
+		_x( 'paid', 'block search term', 'jetpack' ),
+		'patreon',
+		_x( 'pay', 'block search term', 'jetpack' ),
+		_x( 'payments', 'block search term', 'jetpack' ),
+		_x( 'recurring', 'block search term', 'jetpack' ),
+		'stripe',
+		_x( 'subscribe', 'block search term', 'jetpack' ),
```
Comment (mtias) [inline projects/plugins/jetpack/extensions/blocks/donations/index.js:52]: We already have blocks that do this, seems to overrun a bit

File: projects/plugins/jetpack/extensions/blocks/donations/index.js [hunk 6]
```diff
@@ -24,7 +24,38 @@ export const settings = {
...
+		_x( 'Donations', 'block search term', 'jetpack' ),
+		_x( 'earn', 'block search term', 'jetpack' ),
+		_x( 'ecommerce', 'block search term', 'jetpack' ),
+		_x( 'fundraising', 'block search term', 'jetpack' ),
+		_x( 'fundraiser', 'block search term', 'jetpack' ),
+		'gofundme',
+		'go fund me',
+		_x( 'memberships', 'block search term', 'jetpack' ),
+		_x( 'money', 'block search term', 'jetpack' ),
+		_x( 'monthly', 'block search term', 'jetpack' ),
```
Comment (mtias) [inline projects/plugins/jetpack/extensions/blocks/donations/index.js:43]: This seems too generic

File: projects/plugins/jetpack/extensions/blocks/premium-content/index.js [hunk 7]
```diff
@@ -65,10 +65,39 @@ export const settings = {
...
+		_x( 'memberships', 'block search term', 'jetpack' ),
+		_x( 'money', 'block search term', 'jetpack' ),
+		_x( 'monthly', 'block search term', 'jetpack' ),
+		_x( 'newsletter', 'block search term', 'jetpack' ),
+		_x( 'paid', 'block search term', 'jetpack' ),
+		'patreon',
+		_x( 'pay', 'block search term', 'jetpack' ),
+		_x( 'payments', 'block search term', 'jetpack' ),
+		_x( 'paywall', 'block search term', 'jetpack' ),
+		_x( 'premium', 'block search term', 'jetpack' ),
```
Comment (mtias) [inline projects/plugins/jetpack/extensions/blocks/premium-content/index.js:86]: Should be covered by the title

File: projects/plugins/jetpack/extensions/blocks/premium-content/index.js [hunk 8]
```diff
@@ -65,10 +65,39 @@ export const settings = {
...
-		_x( 'membership', 'keyword', 'jetpack' ),
+		_x( 'buy', 'block search term', 'jetpack' ),
+		_x( 'credit card', 'block search term', 'jetpack' ),
+		_x( 'debit card', 'block search term', 'jetpack' ),
+		_x( 'earn', 'block search term', 'jetpack' ),
+		_x( 'exclusive', 'block search term', 'jetpack' ),
+		_x( 'gated', 'block search term', 'jetpack' ),
+		'gofundme',
+		'go fund me',
+		_x( 'login', 'block search term', 'jetpack' ),
```
Comment (mtias) [inline projects/plugins/jetpack/extensions/blocks/premium-content/index.js:76]: Seems farfetched

**Review by jeherve — APPROVED** [review]

**Comment by dsas** [comment]
r240940-wpcom

**Pull request merged**